overlord/snapstate/backend/setup.go

```diff
@@ -25,18 +25,26 @@ import (
 	"path/filepath"
 
 	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/progress"
 	"github.com/snapcore/snapd/release"
 	"github.com/snapcore/snapd/snap"
 )
 
+// InstallUndoContext keeps information about what needs to be undone in case of install failure
+type InstallUndoContext struct {
```

Comment:
I'm not sure Context is a good term here. Maybe InstallRecord ?

Reply:
Agreed, renamed.